File: bindings/csharp/Sop.CLI/ErasureCodingConfigDemo.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Sop.CLI
{
    public static class ErasureCodingConfigDemo
    {
        public static void Run()
        {
            Console.WriteLine("--- Erasure Config Demo ---");
            Console.WriteLine("Note: This demo requires running Redis instance on localhost.");

            // Tell SOP to turn on logging to std err(console).
            Logger.Configure(LogLevel.Info);

            try 
            {
                // Create Clustered Database
                using var ctx = new Context();
                string dbPath = "data/ec_demo";

                var ec = new Dictionary<string, ErasureCodingConfig>
                {
                    {
                        "",
                        new ErasureCodingConfig
                        {
                            DataShards = 1,
                            ParityShards = 1,
                            BaseFolderPathsAcrossDrives = new string[] { dbPath + "/d1", dbPath + "/d2" }
                        }
                    }
                };

                Console.WriteLine($"Creating Clustered Database at {dbPath}...");
                var db = new Database(new DatabaseOptions
                {
                    StoresFolders = new List<string> { dbPath },
                    Type = (int)DatabaseType.Clustered,
                    RedisConfig = new RedisConfig { Address = "localhost:6379" },
                    ErasureConfig = ec,
                });

                Console.WriteLine("Starting Transaction...");
                using (var trans = db.BeginTransaction(ctx))
                {
                    Console.WriteLine("Creating B-Tree 'cluster_btree'...");
                    var btree = db.NewBtree<string, string>(ctx, "cluster_btree", trans);
                    
                    Console.WriteLine("Adding item...");
                    btree.Add(ctx, new Item<string, string>("key1", "value1"));
                    
                    Console.WriteLine("Committing...");
                    trans.Commit();
                }
                
                Console.WriteLine("Clustered operation successful.");
                // Remote the Btree store to cleanup getting ready for next run.
                db.RemoveBtree(ctx, "cluster_btree");
            }
            catch (Exception e)
            {
                Console.WriteLine($"Clustered demo failed (expected if services are not running): {e.Message}");
            }
             Console.WriteLine("--- End of Erasure Coding Demo ---");
        }
    }
}

File: bindings/python/sop/examples/erasure_coding_config_demo.py
import os
import sys

# Add the parent directory to sys.path to import sop
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))

from sop import Context, Logger, LogLevel
from sop.database import Database, DatabaseOptions, DatabaseType
from sop.transaction import ErasureCodingConfig, RedisCacheConfig
from sop.btree import Item

def main():
    print("--- Erasure Coding Config Demo ---")
    print("Note: This demo requires running Redis instance on localhost.")

    # Tell SOP to turn on logging to std err(console).
    Logger.configure(LogLevel.Info)

    try:
        # Create Clustered Database
        ctx = Context()
        db_path = "data/ec_demo"

        ec = {
            "": ErasureCodingConfig(
                data_shards_count=1,
                parity_shards_count=1,
                base_folder_paths_across_drives=[db_path + "/d1", db_path + "/d2"]
            )
        }

        print(f"Creating Clustered Database at {db_path}...")
        db = Database(DatabaseOptions(
            stores_folders=[db_path],
            type=DatabaseType.Clustered,
            redis_config=RedisCacheConfig(address="localhost:6379"),
            erasure_config=ec
        ))

        print("Starting Transaction...")
        with db.begin_transaction(ctx) as trans:
            print("Creating B-Tree 'cluster_btree'...")
            btree = db.new_btree(ctx, "cluster_btree", trans)
            
            print("Adding item...")
            btree.add(ctx, Item(key="key1", value="value1"))
            
            print("Committing...")
            # Commit happens automatically on exit of context manager
        
        print("Clustered operation successful.")
        # Remove the Btree store to cleanup getting ready for next run.
        db.remove_btree(ctx, "cluster_btree")
        
    except Exception as e:
        print(f"Clustered demo failed (expected if services are not running): {e}")
    
    print("--- End of Erasure Coding Demo ---")

if __name__ == "__main__":
    main()
